dgraph/cmd/zero/http.go

```
@@ -231,9 +234,44 @@ func (st *state) getState(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *Server) zeroHealth(ctx context.Context) (*api.Response, error) {
+	if ctx.Err() != nil {
+		return nil, ctx.Err()
+	}
+
+	healthAll := []pb.HealthInfo{pb.HealthInfo{
+		Instance:    "zero",
+		Address:     x.WorkerConfig.MyAddr,
+		Status:      "healthy",
+		Version:     x.Version(),
+		Uptime:      int64(time.Since(x.WorkerConfig.StartTime) / time.Second),
+		LastEcho:    time.Now().Unix(),
+	}}
+
+	jsonOut, err := json.Marshal(healthAll)
+	if err != nil {
+		return nil, errors.Errorf("Unable to Marshal. Err %v", err)
+	}
+	return &api.Response{Json: jsonOut}, nil
+}
+
 func (st *state) pingResponse(w http.ResponseWriter, r *http.Request) {
+	var err error
 	x.AddCorsHeaders(w)
 
-	w.WriteHeader(http.StatusOK)
-	_, _ = w.Write([]byte("OK"))
+	switch r.Header.Get("Accept") {
+		case "application/json":
+			var resp *api.Response
+			if resp, err = (st.zero).zeroHealth(r.Context()); err != nil {
+				x.SetStatus(w, x.Error, err.Error())
+				return
+			}       
+			w.Header().Set("Content-Type", "application/json; charset=utf-8")
+			w.WriteHeader(http.StatusOK)
+			_, _ = w.Write(resp.Json)
```

we shouldn't ignore the error here.